plugins/mcp/mcp-server/src/hopeit_agents/mcp_server/server/mcp.py

```diff
@@ -46,9 +47,30 @@ def run_app(
     workers: int,
     worker_class: str,
     worker_timeout: int,
+    transport: str = "http",
 ) -> None:
-    """Start the MCP HTTP server using the provided runtime configuration."""
+    """Start the MCP server using the provided runtime configuration."""
     init_logger()
+    transport_name = transport.lower()
+
+    if transport_name == "stdio":
+        logger.info(__name__, "Starting MCP Server (transport=stdio).")
+        try:
+            asyncio.run(
+                _serve_stdio(
+                    config_files=config_files,
+                    enabled_groups=enabled_groups,
+                    start_streams=start_streams,
+                )
+            )
+        except KeyboardInterrupt:  # pragma: no cover - manual interrupt
+            logger.info(__name__, "Received interruption, shutting down...")
+        logger.info(__name__, "Stopped MCP Server.")
+        return
+
+    if transport_name != "http":
```

Comment:
To make it more readable so it does not look that is patched with stdio on top, what about using if ... elfif transport_name=="http" and leave this exception for a final else?